src/cargo/util/context/mod.rs

```diff
@@ -2620,6 +2629,20 @@ pub struct CargoBuildConfig {
     // deprecated alias for artifact-dir
     pub out_dir: Option<ConfigRelativePath>,
     pub artifact_dir: Option<ConfigRelativePath>,
+    pub warnings: Option<WarningHandling>,
+}
+
+/// Whether warnings should warn, be allowed, or cause an error.
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Deserialize, Default)]
+#[serde(rename_all = "kebab-case")]
+pub enum WarningHandling {
+    #[default]
+    /// Output warnings.
+    Warn,
+    /// Allow warnings (do not output them).
+    Allow,
+    /// Error if  warnings are emitted.
+    Deny,
```

Comment:
Should we make this `Ord` and switch `build_runner.bcx.gctx.warning_handling()? != WarningHandling::Allow;` to `build_runner.bcx.gctx.warning_handling()? > WarningHandling::Allow;`?

(really unsure, looking for thoughts)

Reply:
Since `LintLevel`, `TomlLintLevel`, and `annotate_snippets::Level` do not impl `Ord`, I think I'd rather not start doing it here. We can always add it later.